### DISCUSSION TITLE
Email Interception for Playwright Testing?

### DISCUSSION THREAD

**Discussion by MoSattler** [epicweb-dev/epic-stack#405] [Q&A] [answered]
I appreciate the resend mock provided, it get's the job done quite nicely in development. I'm wondering though on how to intercept emails during Playwright tests. Specifically, I want to validate the correctness of sign-up confirmation links and evaluate their behavior upon activation. Any suggestions?

**Comment by kentcdodds** [comment, marked as answer]
The emails are written to the file system so you can read them during tests and do whatever you like. This is how our onboarding test works.

**Reply by MoSattler** [reply]
I must have missed that! Thanks for pointing it out!

**Reply by kentcdodds** [reply]
Sure thing 😃
